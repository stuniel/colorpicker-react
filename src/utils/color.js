export function getColorValues(color, type) {
  let reg
  switch (type) {
    case 'hex':
      reg = /^(#?)([a-f\d]{3}|[a-f\d]{6})$/i
      break
    case 'rgb':
      reg = /^rgb\( ?(0|255|25[0-4]|2[0-4]\d|1\d\d|0?\d?\d), ?(0|255|25[0-4]|2[0-4]\d|1\d\d|0?\d?\d), ?(0|255|25[0-4]|2[0-4]\d|1\d\d|0?\d?\d) ?\)$/
      break
    case 'hsl':
      reg = /^hsl\( ?(0|360|35\d|3[0-4]\d|[12]\d\d|0?\d?\d), ?(0|100|\d{1,2})%, ?(0|100|\d{1,2})%\)$/
      break
    case 'r':
    case 'g':
    case 'b':
      reg = /(0|255|25[0-4]|2[0-4]\d|1\d\d|0?\d?\d)/
      break
    default:
      break
  }

  return reg.exec(color)
}

export function formatHEXValue(hex) {
  const value = hex[2]
  const color = value.length === 3
    ? value[0].repeat(2) + value[1].repeat(2) + value[2].repeat(2)
    : value

  return `#${color.toUpperCase()}`
}

export function formatRGBValue(rgb) {
  const [value, r, g, b] = rgb
  return `rgb(${r}, ${g}, ${b})`
}

export function formatHSLValue(hsl) {
  const [value, h, s, l] = hsl
  return `hsl(${h}, ${s}%, ${l}%)`
}

export function formatRGB(input) {
  const [value, r, g, b] = input
  const rgb = {}
  rgb.r = parseFloat(r)
  rgb.g = parseFloat(g)
  rgb.b = parseFloat(b)
  rgb.value = formatRGBValue(input)

  return { ...rgb }
}

export function formatHSL(input) {
  const [value, h, s, l] = input
  const hsl = {}
  hsl.h = parseFloat(h)
  hsl.s = parseFloat(s / 100)
  hsl.l = parseFloat(l / 100)
  hsl.value = formatHSLValue(input)

  return { ...hsl }
}

export function hexToHSL(hex) {
  const result = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(hex.value)
  let r
  let g
  let b
  r = parseInt(result[1], 16)
  g = parseInt(result[2], 16)
  b = parseInt(result[3], 16)

  r /= 255
  g /= 255
  b /= 255

  const max = Math.max(r, g, b)
  const min = Math.min(r, g, b)
  let h
  let s
  const l = (max + min) / 2

  if (max === min) {
    h = 0
    s = 0
  } else {
    const d = max - min
    s = l > 0.5 ? d / (2 - max - min) : d / (max + min)
    switch (max) {
      case r:
        h = ((g - b) / d) + (g < b ? 6 : 0)
        break
      case g:
        h = ((b - r) / d) + 2
        break
      case b:
        h = ((r - g) / d) + 4
        break
      default:
        break
    }
    h /= 6
  }
  const hsl = {}
  hsl.h = Math.ceil(parseFloat(h * 360))
  hsl.s = s
  hsl.l = l
  hsl.value = `hsl(${hsl.h}, ${Math.ceil(parseFloat(s * 100))}%, ${Math.ceil(parseFloat(l * 100))}%)`

  return { ...hsl }
}

export function hexToRGB(hex) {
  const result = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(hex.value)

  const rgb = {}
  rgb.r = parseInt(result[1], 16)
  rgb.g = parseInt(result[2], 16)
  rgb.b = parseInt(result[3], 16)
  rgb.value = `rgb(${rgb.r}, ${rgb.g}, ${rgb.b})`

  return { ...rgb }
}

export function rgbToHEX(rgb) {
  const { r, g, b } = rgb
  const toHex = (x) => {
    const hex = Math.round(x).toString(16).toUpperCase()
    return hex.length === 1 ? `0${hex}` : hex
  }

  const hex = {}
  hex.value = `#${toHex(r)}${toHex(g)}${toHex(b)}`
  return { ...hex }
}

export function hslToRGB(hsl) {
  const { h, s, l } = hsl
  let r
  let g
  let b

  if (s === 0) {
    r = g = b = l
  } else {
    const hueToRGB = (p, q, t) => {
      if (t < 0) { t += 1 }
      if (t > 1) { t -= 1 }
      if (t < 1 / 6) { return p + ((q - p) * 6 * t) }
      if (t < 1 / 2) { return q }
      if (t < 2 / 3) { return p + ((q - p) * ((2 / 3) - t) * 6) }
      return p
    }

    const q = l < 0.5 ? l * (1 + s) : (l + s) - (l * s)
    const p = (2 * l) - q
    r = hueToRGB(p, q, h + (1 / 3))
    g = hueToRGB(p, q, h)
    b = hueToRGB(p, q, h - (1 / 3))
  }

  return [Math.round(r * 255), Math.round(g * 255), Math.round(b * 255)]
}

export function hslToHSV(hsl) {
  const { h, s, l } = hsl

  const hsv = {}
  hsv.h = h
  const t = s * ((l < 0.5) ? l : (1 - l))
  hsv.v = l + t
  hsv.s = (l > 0) ? (2 * t) / hsv.v : hsv.s

  return { ...hsv }
}

export function buildColor(value, type, prevColor) {
  let color
  let colorValue
  let hex
  let hsl
  let hsv
  let rgb
  switch (type) {
    case 'hex':
      colorValue = formatHEXValue(value)
      hex = {}
      hex.value = colorValue
      hsl = hexToHSL(hex)
      rgb = hexToRGB(hex)
      hsv = hslToHSV(hsl)
      break
    case 'rgb':
      rgb = formatRGB(value)
      hex = rgbToHEX(rgb)
      hsl = hexToHSL(hex)
      hsv = hslToHSV(hsl)
      break
    case 'hsl':
      hsl = formatHSL(value)
      hsv = hslToHSV(hsl)
      rgb = hslToRGB(hsl)
      hex = rgbToHEX(rgb)
      break
    case 'r':
      color = prevColor
      color.rgb.r = value[1]
      rgb = color.rgb
      hex = rgbToHEX(rgb)
      hsl = hexToHSL(hex)
      hsv = hslToHSV(hsl)
      break
    case 'g':
      color = prevColor
      color.rgb.g = value[1]
      rgb = color.rgb
      hex = rgbToHEX(rgb)
      hsl = hexToHSL(hex)
      hsv = hslToHSV(hsl)
      break
    case 'b':
      color = prevColor
      color.rgb.b = value[1]
      rgb = color.rgb
      hex = rgbToHEX(rgb)
      hsl = hexToHSL(hex)
      hsv = hslToHSV(hsl)
      break
    default:
      break
  }

  return {
    hsl, hsv, hex, rgb
  }
}

function detectColorType(color) {
  let reg

  reg = /^(#?)([a-f\d]{3}|[a-f\d]{6})$/i
  if (reg.test(color)) { return 'hex' }

  reg = /^rgb\( ?(0|255|25[0-4]|2[0-4]\d|1\d\d|0?\d?\d), ?(0|255|25[0-4]|2[0-4]\d|1\d\d|0?\d?\d), ?(0|255|25[0-4]|2[0-4]\d|1\d\d|0?\d?\d) ?\)$/
  if (reg.test(color)) { return 'rgb' }

  reg = /^hsl\( ?(0|360|35\d|3[0-4]\d|[12]\d\d|0?\d?\d), ?(0|100|\d{1,2})%, ?(0|100|\d{1,2})%\)$/
  if (reg.test(color)) { return 'hsl' }

  return null
}

export function createColorObject(value) {
  let color = value
  let type = detectColorType(color)

  if (type == null) {
    color = '#FFFFFF'
    type = 'hex'
  }

  const colorValues = getColorValues(color, type)
  return buildColor(colorValues, type)
}

export function formatInputValue(color, type) {
  let value
  let values

  switch (type) {
    case 'hex':
      values = getColorValues(color[type].value, type)
      value = formatHEXValue(values)
      break
    case 'rgb':
      values = getColorValues(color[type].value, type)
      value = formatRGBValue(values)
      break
    case 'r':
      value = color.rgb.r
      break
    case 'g':
      value = color.rgb.g
      break
    case 'b':
      value = color.rgb.b
      break
    default:
      break
  }

  return value
}
