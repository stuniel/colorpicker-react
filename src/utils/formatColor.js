function hexToRgb(hex) {
  const result = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(hex.value)

  const r = parseInt(result[1], 16)
  const g = parseInt(result[2], 16)
  const b = parseInt(result[3], 16)
  const value = `rgb(${r}, ${g}, ${b})`

  return { r, g, b, value } // eslint-disable-line object-curly-newline
}

function hslToRgb(hsl) {
  let { h, s, l } = hsl
  let r
  let g
  let b

  h /= 360

  if (s === 0) {
    r = g = b = l // eslint-disable-line no-multi-assign
  } else {
    const hueToRGB = (p, q, t) => {
      if (t < 0) { t += 1 }
      if (t > 1) { t -= 1 }
      if (t < 1 / 6) { return p + ((q - p) * 6 * t) }
      if (t < 1 / 2) { return q }
      if (t < 2 / 3) { return p + ((q - p) * ((2 / 3) - t) * 6) }
      return p
    }

    const q = l < 0.5 ? l * (1 + s) : (l + s) - (l * s)
    const p = (2 * l) - q
    r = hueToRGB(p, q, h + (1 / 3))
    g = hueToRGB(p, q, h)
    b = hueToRGB(p, q, h - (1 / 3))
  }

  r = Math.round(r * 255)
  g = Math.round(g * 255)
  b = Math.round(b * 255)
  const value = `rgb(${r}, ${g}, ${b})`

  return { r, g, b, value } // eslint-disable-line object-curly-newline
}

function hslToHsv(hsl) {
  const { h, s, l } = hsl

  const hsv = {}
  hsv.h = h
  const t = s * ((l < 0.5) ? l : (1 - l))
  hsv.v = l + t
  hsv.s = (l > 0) ? (2 * t) / hsv.v : hsl.s

  return { ...hsv }
}

function rgbToHex(rgb) {
  const { r, g, b } = rgb
  const toHex = (x) => {
    const hex = Math.round(x).toString(16).toUpperCase()
    return hex.length === 1 ? `0${hex}` : hex
  }

  const value = `#${toHex(r)}${toHex(g)}${toHex(b)}`
  return { value }
}

function rgbToHsl(rgb) {
  let { r, g, b } = rgb
  r /= 255
  g /= 255
  b /= 255

  const max = Math.max(r, g, b)
  const min = Math.min(r, g, b)
  let h
  let s
  const l = (max + min) / 2

  if (max === min) {
    h = s = 0 // eslint-disable-line no-multi-assign
  } else {
    const d = max - min
    s = l > 0.5 ? d / (2 - max - min) : d / (max + min)

    switch (max) {
      case r: h = (g - b) / d + (g < b ? 6 : 0); break
      case g: h = (b - r) / d + 2; break
      case b: h = (r - g) / d + 4; break
      default: h = 0
    }
  }

  h *= 60

  const value = `hsl(${Math.round(h)}, ${Math.round(s * 100)}%, ${Math.round(l * 100)}%)`

  return { h, s, l, value } // eslint-disable-line object-curly-newline
}

function rToRGB(r, prevColor) {
  const { rgb } = prevColor
  const { g, b } = rgb
  r = parseFloat(r)
  const value = `rgb(${r}, ${g}, ${b})`
  return { ...rgb, r, value }
}

function gToRGB(g, prevColor) {
  const { rgb } = prevColor
  const { r, b } = rgb
  g = parseFloat(g)
  const value = `rgb(${r}, ${g}, ${b})`
  return { ...rgb, g, value }
}

function bToRGB(b, prevColor) {
  const { rgb } = prevColor
  const { r, g } = rgb
  b = parseFloat(b)
  const value = `rgb(${r}, ${g}, ${b})`
  return { ...rgb, b, value }
}

function formatHexValue(hex) {
  const value = hex[2]
  const color = value.length === 3
    ? value[0].repeat(2) + value[1].repeat(2) + value[2].repeat(2)
    : value

  return `#${color.toUpperCase()}`
}

function formatRgbValue(rgb) {
  const [value, r, g, b] = rgb
  return `rgb(${r}, ${g}, ${b})`
}

function formatHslValue(hsl) {
  const [value, h, s, l] = hsl
  return `hsl(${h}, ${s}%, ${l}%)`
}

function formatHex(input) {
  const hex = {}
  hex.value = formatHexValue(input)

  return { ...hex }
}

function formatRgb(input) {
  const [value, r, g, b] = input
  const rgb = {}
  rgb.r = parseFloat(r)
  rgb.g = parseFloat(g)
  rgb.b = parseFloat(b)
  rgb.value = formatRgbValue(input)

  return { ...rgb }
}

function formatHsl(input) {
  const [value, h, s, l] = input
  const hsl = {}
  hsl.h = parseFloat(h)
  hsl.s = parseFloat(s / 100)
  hsl.l = parseFloat(l / 100)
  hsl.value = formatHslValue(input)

  return { ...hsl }
}

const colors = {
  hex: {
    reg: /^(#?)([a-f\d]{3}|[a-f\d]{6})$/i,
    rgb: hexToRgb,
    value: formatHex
  },
  rgb: {
    reg: /^rgb\( ?(0|255|25[0-4]|2[0-4]\d|1\d\d|0?\d?\d), ?(0|255|25[0-4]|2[0-4]\d|1\d\d|0?\d?\d), ?(0|255|25[0-4]|2[0-4]\d|1\d\d|0?\d?\d) ?\)$/,
    rgb: rgb => rgb,
    hex: rgbToHex,
    hsl: rgbToHsl,
    value: formatRgb
  },
  r: {
    reg: /^([0-9]|[1-9]\d|1\d\d|2[0-4]\d|25[0-5])$/,
    rgb: rToRGB,
    value: r => r
  },
  g: {
    reg: /^([0-9]|[1-9]\d|1\d\d|2[0-4]\d|25[0-5])$/,
    rgb: gToRGB,
    value: g => g
  },
  b: {
    reg: /^([0-9]|[1-9]\d|1\d\d|2[0-4]\d|25[0-5])$/,
    rgb: bToRGB,
    value: b => b
  },
  hsl: {
    reg: /^hsl\( ?(0|360|35\d|3[0-4]\d|[12]\d\d|0?\d?\d), ?(0|100|\d{1,2})%, ?(0|100|\d{1,2})%\)$/,
    rgb: hslToRgb,
    hsv: hslToHsv,
    value: formatHsl
  }
}

export {
  hexToRgb,
  hslToRgb,
  rgbToHex,
  formatHexValue,
  formatRgbValue,
  formatHslValue,
  formatRgb,
  formatHsl,
  hslToHsv,
  colors
}
