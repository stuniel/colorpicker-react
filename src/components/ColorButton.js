import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import isEqual from 'lodash.isequal'

import { createColorObject } from '../utils/color'
import withColor from './withColor'

const propTypes = {
  color: PropTypes.shape({
    hsl: PropTypes.shape({
      h: PropTypes.number,
      s: PropTypes.number,
      l: PropTypes.number,
      value: PropTypes.string
    }),
    hsv: PropTypes.shape({
      h: PropTypes.number,
      s: PropTypes.number,
      v: PropTypes.number,
      value: PropTypes.string
    }),
    rgb: PropTypes.shape({
      r: PropTypes.number,
      g: PropTypes.number,
      b: PropTypes.number,
      value: PropTypes.string
    }),
    hex: PropTypes.shape({
      value: PropTypes.string
    })
  }).isRequired,
  onChange: PropTypes.func.isRequired,
  value: PropTypes.string.isRequired,
  radius: PropTypes.string,
  style: PropTypes.object // eslint-disable-line react/forbid-prop-types
}
const defaultProps = {
  radius: '2em',
  style: {}
}

const Button = styled.button`
  width: ${props => props.radius};
  height: ${props => props.radius};
  border: ${props => `3px solid ${props.value}`};
  border-radius: ${props => props.radius};
  background-color: ${props => (props.active ? 'transparent' : props.value)};
  outline: none;
  box-sizing: border-box;
  transition: 0.4s all cubic-bezier(0.85, 0, 0.15, 1);
  cursor: pointer;
`

class ColorButton extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      active: isEqual(props.color, createColorObject(props.value))
    }
    this.handleMouseDown = this.handleMouseDown.bind(this)
  }

  componentDidUpdate(prevProps) {
    if (prevProps.color !== this.props.color || prevProps.value !== this.props.value) {
      const active = isEqual(this.props.color, createColorObject(this.props.value))
      this.setState({ active }) // eslint-disable-line react/no-did-update-set-state
    }
  }

  handleMouseDown() {
    const { onChange, value } = this.props

    const color = createColorObject(value)

    if (onChange) {
      onChange(color)
    }
  }

  render() {
    const { radius, value, style } = this.props
    const { active } = this.state

    return (
      <Button
        active={active}
        onMouseDown={this.handleMouseDown}
        radius={radius}
        style={style}
        value={value}
      />
    )
  }
}

ColorButton.propTypes = propTypes
ColorButton.defaultProps = defaultProps

export default withColor(ColorButton)
