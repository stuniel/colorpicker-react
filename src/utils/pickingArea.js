import { hslToRgb, rgbToHex } from './color'

function getPosition(color) {
  const { hsv: { s, v } } = color

  const top = `${(1 - v) * 100}%`
  const left = `${s * 100}%`

  return { top, left }
}

function getColorFromHsv(h, s, v) {
  const hsv = {}
  hsv.h = h
  hsv.s = s
  hsv.v = v
  hsv.value = `hsv(${Math.round(hsv.h)}, ${(hsv.s * 100 | 0)}%, ${(hsv.v * 100 | 0)}%)`

  const hsl = {}
  hsl.h = hsv.h
  hsl.l = (2 - hsv.s) * (hsv.v / 2)
  hsl.s = hsl.l < 1
    ? (hsv.s * hsv.v) / (hsl.l < 0.5 ? hsl.l * 2 : 2 - (hsl.l * 2))
    : hsl.s
  hsl.s = Number.isNaN(hsl.s) ? 0 : hsl.s
  hsl.value = `hsl(${Math.round(hsl.h)}, ${(hsl.s * 100 | 0)}%, ${(hsl.l * 100 | 0)}%)`

  const rgb = hslToRgb(hsl)
  const hex = rgbToHex(rgb)

  return { hex, rgb, hsl, hsv }
}

function getColor(event, color, parent) {
  const { hsv: { h } } = color
  const rect = parent.getBoundingClientRect()

  let left = (event.targetTouches && event.targetTouches[0].clientX) || event.clientX
  let top = (event.targetTouches && event.targetTouches[0].clientY) || event.clientY

  left -= rect.left
  top -= rect.top

  left = Math.min(Math.max(left, 0), rect.width)
  top = Math.min(Math.max(top, 0), rect.height)

  const saturation = left / rect.width
  const value = 1 - (top / rect.height)

  return getColorFromHsv(h, saturation, value)
}

function getColorFromKeyDown(color, key) {
  const { hsv: { h, s, v } } = color
  const step = 0.01

  const x = (key === 39 && step) || (key === 37 && -(step)) || 0
  const y = (key === 40 && step) || (key === 38 && -(step)) || 0

  const saturation = Math.min(Math.max((s + x), 0), 1)
  const value = Math.min(Math.max((v - y), 0), 1)

  return getColorFromHsv(h, saturation, value)
}

export {
  getPosition,
  getColor,
  getColorFromKeyDown
}
