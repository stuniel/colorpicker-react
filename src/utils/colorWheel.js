import { hslToHsv, hslToRgb, rgbToHex } from './formatColor'

function radiansToDegrees(radians) {
  return (radians * 180) / Math.PI
}

function getPosition(color, parent) {
  const { hsl: { h, l } } = color
  const { width, height } = parent.getBoundingClientRect()
  const RADIUS = 50 // 50% of parent's width/height
  const radius = width / 2
  const angle = (h - 90) / 360

  const x = Math.cos((angle) * Math.PI * 2)
  const y = Math.sin((angle) * Math.PI * 2)

  let left = ((radius + (radius * x * (1 - l))) / width) * 100
  let top = ((radius + (radius * y * (1 - l))) / height) * 100

  const xLength = left - RADIUS
  const yLength = top - RADIUS

  const alpha = (Math.atan2(yLength, xLength))
  const length = Math.sqrt((Math.abs(xLength)**2) + (Math.abs(yLength)**2))

  if (length > RADIUS) {
    left = (Math.cos(alpha) * RADIUS) + RADIUS
    top = (Math.sin(alpha) * RADIUS) + RADIUS
  }

  left = `${left}%`
  top = `${top}%`

  return { left, top }
}

function getColorFromCoordinates(left, top, x, y, radius) {
  const angle = radiansToDegrees(Math.atan2(y, x)) + 90

  const alpha = (Math.atan2(y, x))
  let length = Math.sqrt(Math.abs(x)**2 + Math.abs(y)**2)

  if (length > radius) {
    left = (Math.cos(alpha) * radius) + radius
    top = (Math.sin(alpha) * radius) + radius
    length = radius
  }

  const hsl = {}
  hsl.h = angle < 0 ? angle + 360 : angle
  hsl.l = 1 - (length / radius)
  hsl.s = 1
  hsl.value = `hsl(${Math.round(hsl.h)}, ${(hsl.s * 100 | 0)}%, ${(hsl.l * 100 | 0)}%)`

  const hsv = hslToHsv(hsl)
  const rgb = hslToRgb(hsl)
  const hex = rgbToHex(rgb)

  return { hex, rgb, hsl, hsv }
}

function getColor(event, color, parent) {
  const rect = parent.getBoundingClientRect()

  let left = (event.targetTouches && event.targetTouches[0].clientX) || event.clientX
  let top = (event.targetTouches && event.targetTouches[0].clientY) || event.clientY

  left -= rect.left
  top -= rect.top

  const radius = rect.width / 2
  const x = left - radius
  const y = top - radius

  return getColorFromCoordinates(left, top, x, y, radius)
}


function getColorFromKeyDown(color, key, parent) {
  let { top, left } = getPosition(color, parent)

  top = parseFloat(top)
  left = parseFloat(left)

  const step = 1

  const stepX = (key === 39 && step) || (key === 37 && -(step)) || 0
  const stepY = (key === 40 && step) || (key === 38 && -(step)) || 0

  const radius = 50
  const x = (left + stepX) - 50
  const y = (top + stepY) - 50

  return getColorFromCoordinates(left, top, x, y, radius)
}

export {
  getPosition,
  getColor,
  getColorFromKeyDown
}
