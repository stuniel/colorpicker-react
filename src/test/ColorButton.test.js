import React from 'react'
import { mount, shallow } from 'enzyme'

import ColorButton from '../components/ColorButton'
import { createColorObject } from '../utils/color'

const createShallowWrapper = (props, context) => {
  const outer = shallow(<ColorButton {...props} />)
  const Component = outer.props().children(context)

  return shallow(Component)
}

const createMountWrapper = (props, context) => {
  const outer = shallow(<ColorButton {...props} />)
  const Component = outer.props().children(context)

  return mount(Component)
}

describe('<ColorButton />', () => {
  const context = {
    color: createColorObject('#FFFFFF'),
    onChange: () => {}
  }

  it('renders correctly', () => {
    const wrapper = createShallowWrapper({ value: '#000000' }, context)
    expect(wrapper).toMatchSnapshot()
    wrapper.unmount()
  })
})

describe('handleMouseDown', () => {
  it('passes \'onChange\' correctly', () => {
    const onChange = jest.fn()

    const context = {
      color: createColorObject('#FFFFFF'),
      onChange
    }

    const wrapper = createMountWrapper({ value: '#000000' }, context)
    const button = wrapper.find('button')
    button.simulate('mousedown')

    expect(onChange).toHaveBeenCalledTimes(1)
    wrapper.unmount()
  })
})
