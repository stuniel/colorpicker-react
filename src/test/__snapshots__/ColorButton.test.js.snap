// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ColorButton /> renders correctly 1`] = `
<styled.button
  active={false}
  onMouseDown={[Function]}
  radius="2em"
  style={Object {}}
  value="#000000"
/>
`;
